src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -331,12 +334,13 @@ private static List<AssetTable> processAssetTables(
   }
 
   private static List<AssetRelationship> processAssetRelationships(
-      List<String> assetRelationshipNames, Map<String, Relationship> relationships) {
-    return Collections.unmodifiableList(
-        relationships.entrySet().stream()
-            .filter(map -> assetRelationshipNames.contains(map.getKey()))
-            .map(entry -> new AssetRelationship().datasetRelationship(entry.getValue()))
-            .collect(Collectors.toList()));
+      List<String> orderedAssetRelationshipNames, Map<String, Relationship> relationships) {
+    return orderedAssetRelationshipNames.stream()
+        .filter(relationships::containsKey)
+        .map(
+            relationship ->
+                new AssetRelationship().datasetRelationship(relationships.get(relationship)))
```

Comment:
Since the lambda type is `String` (and not `Relationship`) I would call this `name`
```suggestion
            name ->
                new AssetRelationship().datasetRelationship(relationships.get(name)))
```